quantum/rgblight.c

```diff
@@ -95,11 +95,19 @@ LED_TYPE led[RGBLED_NUM];
 #    define LED_ARRAY led
 #endif
 
+#ifndef RGBLIGHT_CUSTOM_DRIVER
 static uint8_t clipping_start_pos = 0;
 static uint8_t clipping_num_leds  = RGBLED_NUM;
 static uint8_t effect_start_pos   = 0;
 static uint8_t effect_end_pos     = RGBLED_NUM;
 static uint8_t effect_num_leds    = RGBLED_NUM;
+#else
```

Comment:
Is there an existing use case for this change? If not, I would wait for that rather than a preemptive change. If so, I would rather see getters than externs, especially as the duplication is rather messy.

Reply:
Yeah, probably. 

Specifically, this block is used in `rgblight_set`, and is for the clipping range stuff.   However, if you enable the custom driver option, `rgblight_set` is no longer defined in this file, and because these are static, they're not accessible outside of this file (or that's my understanding, at least). 

Since I would like to maintain that, I wasn't exactly sure how to handle this.  

Would it be better as a separate PR?  And would it be better to have them not static, and extern'd normally?

Reply:
I get the codes use within this block, that description is implementation, not a use case.

However if you said you wanted to use RGBLED_SPLIT, and that depends on clipping, then it would be.

I dont see the point in making the code dirty for a "probably".

Reply:
It's not just used for rgbled split, though. It's used by `rgblight_set_clipping_range(x, y)`, as well.  Which is, yes, used by the `RGBLED_SPLIT` functionality by default, but not exclusively. 

For now, I'll revert this change, and pull out the clipping functionality out of the ergodox's code, for simplicity.   And open a PR/issue/whatever to discuss this in more detail.